### PR TITLE
Add scripts entry points for subpackages

### PR DESCRIPTION
**Pull request by ijpulidos**
## Description
These changes add scripts entry points for the scripts in the `scripts` directory of every subpackage.

While doing so it also clean ups the scripts themselves, avoiding having to manipulate the python system path.

In the long run we probably don't want as many scripts but rather some centralized CLI entry point that runs subcommands for each of these scripts. But for now this will do, since these need to be used by the researchers asap.

## Todos
Notable points that this PR has either accomplished or will accomplish.
  - [x] scripts for data subpackage
  - [x] scripts for dataviz subpackage
  - [x] scripts for docking subpackage
  - [x] scripts for ml subpackage
  - [x] scripts for modeling subpackage
  - [ ] scripts for simulation subpackage

## Status
- [x] Ready to go

### CODE REVIEW

**Comment by ijpulidos** [comment]
@kaminow @JenkeScheen @apayne97 As we discussed, this should be ready to be reviewed but there are two things we want to review first before merging:

- [ ] Clean up. Are all these scripts needed? If not, we can remove the respective entries from `pyproject.toml`.
- [ ] Some scripts (especially in the `dataviz` subpackage) don't have a `main` callable, if we need these scripts to be accessible then we need to point to some "main-like" callable in them.

If you need to check which scripts are included and need review, you can just check the `[project.scripts]` section on the `pyproject.toml` file for each subpackage. Only what's included there will be accessible as a script (without the need to change any `sys.path` variables). Let me know if that works, thanks!

**Comment by ijpulidos** [comment]
Just to be clear, we just probably need your feedback on what scripts are required (or which aren't), and if there are better names for them (for example the script `train` in the `ml` subpackage could use a better name). I can make the changes once we agree on these.

**Comment by ijpulidos** [comment]
@apayne97 Since we are now accessing these scripts from "anywhere" I removed the `repo_path` trick and its references, especially in https://github.com/choderalab/asapdiscovery/pull/150/commits/43eabca98dce1d4b46d09580c5b873f542bd6b8c . They are now stored by default wherever the working directory is when running the scripts, does this seem ok?

**Comment by ijpulidos** [comment]
Thanks! I see csv file with the scripts that we want, I'm gonna make sure that we have those in our entry points and move forward to trying to get this merged into main (solving conflicts, etc.). 

I'll be removing the csv from the repo (just to not have another file to sync) and put its contents here for future reference.

These are the scripts we want to keep to this date:

```csv
# 0: move to ./attic/, 1:request help if this needs to be kept, 2: keep
2,./asapdiscovery-dataviz/asapdiscovery/dataviz/scripts/color_active_site.pml
0,./asapdiscovery-dataviz/asapdiscovery/dataviz/scripts/dash-all-results-exploration.py
2,./asapdiscovery-dataviz/asapdiscovery/dataviz/scripts/dash-docking-contour-plot.py
0,./asapdiscovery-dataviz/asapdiscovery/dataviz/scripts/dash-docking-exploration.py
2,./asapdiscovery-dataviz/asapdiscovery/dataviz/scripts/dash-docking-interactive-scatterplots.py
2,./asapdiscovery-dataviz/asapdiscovery/dataviz/scripts/dash-docking-interactive-table.py
1,./devtools/scripts/create_conda_env.py
2,./asapdiscovery-data/asapdiscovery/data/scripts/cdd_to_schema.py
2,./asapdiscovery-data/asapdiscovery/data/scripts/create_mers_schema.py
2,./asapdiscovery-data/asapdiscovery/data/scripts/docking_data_to_csv.py
2,./asapdiscovery-data/asapdiscovery/data/scripts/download_fragalysis_data.py
2,./asapdiscovery-data/asapdiscovery/data/scripts/download_moonshot_data.py
2,./asapdiscovery-data/asapdiscovery/data/scripts/download_pdbs.py
2,./asapdiscovery-data/asapdiscovery/data/scripts/filter_fragalysis_by_smarts.py
2,./asapdiscovery-data/asapdiscovery/data/scripts/write_cmpd_to_frag_dict.py
2,./asapdiscovery-ml/asapdiscovery/ml/scripts/pred_comp.py
2,./asapdiscovery-ml/asapdiscovery/ml/scripts/rank_enantiomers.py
2,./asapdiscovery-ml/asapdiscovery/ml/scripts/train.py
2,./asapdiscovery-ml/asapdiscovery/ml/scripts/upset_plot_ranked.py
0,./asapdiscovery-docking/asapdiscovery/docking/scripts/build_design_units.py
2,./asapdiscovery-docking/asapdiscovery/docking/scripts/clean_results_csv.py
2,./asapdiscovery-docking/asapdiscovery/docking/scripts/collate_results.py
2,./asapdiscovery-docking/asapdiscovery/docking/scripts/create_fauxalysis_subset_from_yaml.py
2,./asapdiscovery-docking/asapdiscovery/docking/scripts/fauxalysis_from_docking.py
2,./asapdiscovery-docking/asapdiscovery/docking/scripts/find_all_mcs.py
0,./asapdiscovery-docking/asapdiscovery/docking/scripts/find_mcs_mers.py
2,./asapdiscovery-docking/asapdiscovery/docking/scripts/hallucinate_fragalysis.py
2,./asapdiscovery-docking/asapdiscovery/docking/scripts/make_docked_complexes.py
0,./asapdiscovery-docking/asapdiscovery/docking/scripts/mass_add_seqres.py
2,./asapdiscovery-docking/asapdiscovery/docking/scripts/mcs_res_mpro_to_compound.py
2,./asapdiscovery-docking/asapdiscovery/docking/scripts/plot_docking_results.py
2,./asapdiscovery-docking/asapdiscovery/docking/scripts/prep_fragalysis_dus.py
2,./asapdiscovery-docking/asapdiscovery/docking/scripts/prep_pdb_structures.py
2,./asapdiscovery-docking/asapdiscovery/docking/scripts/prep_proteins.py
2,./asapdiscovery-docking/asapdiscovery/docking/scripts/run_docking_oe.py
2,./asapdiscovery-docking/asapdiscovery/docking/scripts/score_structures.py
2,./asapdiscovery-docking/asapdiscovery/docking/scripts/test-protein-prep.py
2,./asapdiscovery-modeling/asapdiscovery/modeling/scripts/blast_search.py
2,./asapdiscovery-modeling/asapdiscovery/modeling/scripts/copy_and_clean_frag.sh
2,./asapdiscovery-modeling/asapdiscovery/modeling/scripts/make_foldalysis.py
2,./asapdiscovery-modeling/asapdiscovery/modeling/scripts/run_colabfold.sh
```

**Comment by ijpulidos** [comment]
@JenkeScheen Thanks, I think for now I won't remove the scripts (that means, the actual py files). I'd just remove them from the entry points, such that they are not "accessible" but the code will still live in the repo. 

I don't think we are going to be needing the `./devtools/scripts/create_conda_env.py`.

Thanks!

**Comment by ijpulidos** [comment]
@apayne97 @JenkeScheen @kaminow This one should be ready for review now. 

Please note that I had to change some of the names of the script files to be _pythonic_, in the sense of using underscores instead of hyphens for the file names. The script commands do have hyphens, which I think is the convention. That is, if we want to call the `clean_results_csv.py` script, then the call would be `clean-results-csv` on the command line.

File: asapdiscovery-data/asapdiscovery/data/scripts/write_cmpd_to_frag_dict.py [hunk 1]
```diff
@@ -35,8 +30,6 @@ def get_args():
         "-o",
         "--out_yaml",
         default=os.path.join(
-            repo_path,
-            "data",
             "cmpd_to_frag.yaml",
         ),
```
Comment (apayne97) [inline asapdiscovery-data/asapdiscovery/data/scripts/write_cmpd_to_frag_dict.py]: I'm going to change this to no longer be a default path and instead just put a suggested path in the example

**Review by apayne97 — APPROVED** [review]
Just some minor commits for a few scripts where now a default argument doesn't make sense since I was getting the "repo_path" as a variable. Only I have ever used those scripts I think so removing that default argument and instead having it be a required parameter should be fine! (which I've done).

File: asapdiscovery-data/asapdiscovery/data/scripts/write_cmpd_to_frag_dict.py (hunk 1 above)
Comment (ijpulidos) [reply asapdiscovery-data/asapdiscovery/data/scripts/write_cmpd_to_frag_dict.py]: I agree, this is better. Thanks!

**Pull request merged**